src/utils/string.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Compares two string given in lexicographical order.
+ * @returns 1 if `a` is larger than `b`; -1 if `a` is smaller/older than `b`; 0 otherwise (same message)
+ */
+export function lexicographicalCompare(a: string, b: string): number {
```

Comment:
any reason we can't use something like this?
```
a.localeCompare(b, undefined, { numeric: true })
```
(the `numeric: true` makes it so that `"1" < "2" < "10"` instead of `"1" < "10" < "2"`)

Reply:
I didn't use `localeCompare` as the context of the string compare here is not locale dependent and I don't know if there are implications of using `localCompare` in different locales so choose to not bother with it.